scripts/sync-bottle-hashes.sh

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env bash
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# This script takes the new tag and the OS name as its arguments
+# and handles git shenanigans related to syncing two pipelines over a common branch.
+
+set -e
+
+if [ -z "$1" ] || [ -z "$2" ]; then
+    echo "Please call this script with the release tag and the OS name."
+else
+    git config user.name "Serokell CI bot" # necessary for pushing
+    git config user.email "hi@serokell.io"
+    git fetch --all
+
+    branch_name="auto/update-brew-formulae-$1"
+
+    function compute_and_commit {
+        ./scripts/bottle-hashes.sh .
+        git commit -a -m "[Chore] Add $1 hashes to brew formulae"
+    }
+
+    # Git doesn't have an easy way to check out a branch regardless of whether it exists.
+    if ! git switch "$branch_name"; then
+        git switch -c "$branch_name"
+        git push --set-upstream origin "$branch_name"
+    fi
+
+    compute_and_commit "$2"
+
+    set +e
+    git push
+
+    if ! git push; then
+        # If there is a collision precisely at the time of pushing, that means the other pipeline
+        # has pushed. Reset to origin and try again. Once should always be enough.
+        set -e
+        git fetch --all
+        git reset --hard origin/"$branch_name"
+        compute_and_commit "$2"
+        git push
+    else
```

Comment:
> Once should always be enough

Well, only as long as there are no more than 2 bots, which might not always be the case.

Why not make this `if` a `while` instead? 

I believe that in that case the `fetch` and `reset` might be put in the `compute_and_commit` function as well.

Maybe it's even possible to do a `do ... while` loop and not use a function at all (it seems doable with [some tricks](https://stackoverflow.com/a/27761760/10986307)).

P.s. isn't the `git push` at the end here redundant as well?

Reply:
I disagree with turning it into a do-while loop in a way you linked. I tried it and IMO it's very unreadable if you're not an 80th level bash-fu master.
Is the way I tried in the last commit good enough? I think it should work the same.

Reply:
Oh, your solution is perfectly fine IMO :ok_hand: 

I mostly linked to that answer for the explanation :sweat_smile: